sdk/core/ci.yml

```diff
@@ -81,3 +81,9 @@ extends:
     AdditionalModules:
       - name: perf-test-core
         groupId: com.azure
+    MatrixConfigs:
```

Comment:
@benbp is this the best way to add this custom matrix. I don't like that we are duplicating the matrix and it would be great if we didn't need to do that.

Reply:
@weshaggard I'm not a fan of the duplication either. We were trying to figure out a way to do this with the base matrix and were coming up empty, because if we do a high level `replace` operation for `TestGoals` then it will override the `TestGoals` entry when `TestFromSource=true` which is not desired. We don't have a "replace where xyz fields match" query type (and I believe that was an intentional decision to keep the replace syntax simple).

Another approach here could be to split `TestGoals` into two variables, one used in the base matrix, and one used in the includes (like `TestVariantGoals` or something), and then core could replace only `TestGoals`.

Reply:
I feel we have to figure out a good way to share this or we definitely risk the core pipeline having different matrix configurations then the rest of the repo which isn't a good state to be in.